llama_stack/apis/datasets/datasets.py

```diff
@@ -24,7 +24,7 @@ class DatasetDef(BaseModel):
         description="The schema definition for this dataset",
     )
     url: URL
-    metadata: Dict[str, Any] = Field(
+    metadata: Dict[str, str] = Field(
```

Comment:
Can Pydantic's model_dump (https://docs.pydantic.dev/latest/concepts/serialization/) work with `Any`? I previously used it for serializing scoring fns to json in https://github.com/meta-llama/llama-stack/pull/330/commits/8aee752c19a18246dcefdcbb736cf17218573e9f#diff-65400ea46ff38ef0cbf5cad95b4ee92675c9871712351ae4f89542f08ab43967

Reply:
Can you show an example where this didn't work?

Reply:
```
Traceback (most recent call last):
  File "/home/dineshyv/.conda/envs/llamastack-meta-reference-test/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/dineshyv/.conda/envs/llamastack-meta-reference-test/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/dineshyv/local/llama-stack/llama_stack/distribution/server/server.py", line 43, in <module>
    import llama_stack.distribution.store as distribution_store
  File "/home/dineshyv/local/llama-stack/llama_stack/distribution/store/__init__.py", line 1, in <module>
    from .registry import DiskRegistry, Registry
  File "/home/dineshyv/local/llama-stack/llama_stack/distribution/store/registry.py", line 20, in <module>
    DESERIALIZER = create_deserializer(RoutableObjectWithProvider)
  File "/home/dineshyv/local/llama-stack/docs/openapi_generator/strong_typing/deserializer.py", line 819, in create_deserializer
    return _get_deserializer(typ, context)
  File "/home/dineshyv/local/llama-stack/docs/openapi_generator/strong_typing/deserializer.py", line 865, in _get_deserializer
    deserializer.build(context)
  File "/home/dineshyv/local/llama-stack/docs/openapi_generator/strong_typing/deserializer.py", line 378, in build
    self.member_parsers = tuple(
  File "/home/dineshyv/local/llama-stack/docs/openapi_generator/strong_typing/deserializer.py", line 379, in <genexpr>
    _get_deserializer(member_type, context) for member_type in self.member_types
  File "/home/dineshyv/local/llama-stack/docs/openapi_generator/strong_typing/deserializer.py", line 861, in _get_deserializer
    deserializer.build(context)
  File "/home/dineshyv/local/llama-stack/docs/openapi_generator/strong_typing/deserializer.py", line 778, in build
    parser = _get_deserializer(required_type, context)
  File "/home/dineshyv/local/llama-stack/docs/openapi_generator/strong_typing/deserializer.py", line 865, in _get_deserializer
    deserializer.build(context)
  File "/home/dineshyv/local/llama-stack/docs/openapi_generator/strong_typing/deserializer.py", line 267, in build
    self.value_parser = _get_deserializer(self.value_type, context)
  File "/home/dineshyv/local/llama-stack/docs/openapi_generator/strong_typing/deserializer.py", line 864, in _get_deserializer
    deserializer = _create_deserializer(typ)
  File "/home/dineshyv/local/llama-stack/docs/openapi_generator/strong_typing/deserializer.py", line 939, in _create_deserializer
    raise TypeError(f"unable to de-serialize unrecognized type: {typ}")
TypeError: unable to de-serialize unrecognized type: typing.Any
```

Reply:
oh you shouldn't need any DESERIALIZER or anything like that at all -- you should be able to get by with `obj.model_dump_json()` and `Cls(**json.loads(...))`

Reply:
yes, make this an `Annotated` type with a discriminator and then use `pydantic.parse_obj_as()`

Reply:
There are a couple ways to do this: 
- store the type of the object yourself and then switch on it in your custom deserializer 
- do the Annotated thing I suggested above

Reply:
done. added back Any